### ISSUE TITLE
Accounts / Characters

### ISSUE THREAD

**Issue by wtl420** [TheMinusWorld/Mayhem-Whereabouts#4]
We need to establish what we want accounts and characters to do exactly and how that should work. I believe this should be the first thing to focus on as pretty much everything else is going to rely on accounts in some form.

This information should be documented on the Wiki once we've made decisions.

I believe we should have a chart for what features are relevant to accounts and which are relevant to characters.

Once we decide on the defined roles of each we should define some kind of API endpoints for accessing data from each.

**Comment by wtl420** [comment]
Testing webhook

**Comment by DarkoJohnny** [comment]
I like the idea of Characters being distinct from accounts. Accounts' parameters should be limited to just their name/password/e-mail/DoB/joindate, with them being simply what you sign into to gain access to your Character list (With a Character being made automatically with your account's name for ease of use). Characters would have their own individual parameters such as avatar/signature/EX/GSR/postcount/activity what have you, with Characters being able to be swapped between different accounts or be operated under multiple accounts at once as per your own previous suggestion some time back.

Some side notes:
-Good Sh*ts and Reactions should be account locked rather than Character locked for obvious reasons, so that'd be one more thing that should be tracked by accounts other than the bare minimum of information listed prior.
-To handle the potential issue of handling Characters now having their own EX bars/Rally and the annoyances that come from that, Spellcasting should accommodate for this by letting you choose which Character you wish to cast from. Not directly related to this discussion just something I thought about when writing this.

**Comment by keithpickering** [comment]
I'm trying to decide whether there's a good use case for having different EX/Mana (didn't we discuss splitting these at some point?) between characters under the same account. Seems like it'd only really be useful if characters had unique gameplay attributes that affect which spells/items are available, how much they cost, etc. Maybe the Class (Babyface etc) could be character-specific

Edit: actually now that I think about it, that's an exploit. You could just make a character for each class which isn't cool. So I think class as well as EX/Mana should be account-level. Choosing which character to cast a spell as is super important though, we're going to make sure character switching is seamless no matter where you are. I'd like to think about incorporating a floating action button in this design so maybe it could be under that.

**Comment by wtl420** [comment]
I don’t think we need to worry too much about ex yet; I can see that as it’s own feature. I agree we want to find a good balance between keeping it more account based but also having it visible from characters- might be easier to see how we can approach that once we get the basic account structure figured out though.

**Comment by DarkoJohnny** [comment]
EX being Character instead of account based was to accommodate for particular donation type scenarios as well as to make shared Characters more interesting, but I suppose the workarounds necessary might make it easier to just make a new account for donation scenarios and the shared EX pool for certain Characters could be avoided by casting from another Character anyways.

**Comment by FunWithDespair** [comment]
I think making characters for different classes should be a viable tactic. Say that your main account is a Babyface and you made a Wario dupe/character, or I dunno, someone who's just an asshole character. It doesn't really make sense to also have the asshole dupe also function as a Babyface when it would work much better as a Hotshot. It would encourage people to mix up posting styles a bit between dupes, anyway.

**Comment by keithpickering** [comment]
I definitely disagree, to me the point of classes is to weigh the
costs/benefits of each and decide which one you want to take your chances
with. If you can just play all of them interchangeably, we might as well
just let you switch them at any time, in which case there's absolutely no
weight behind the decision. We need to balance convenience with limitations
in order for gameplay to be fun.

**Comment by wtl420** [comment]
I think classes can be figured out separately from accounts / characters. I don’t think there’s anything we need to worry about with classes or ex just yet; I’ll create some separate issues for things in a brainstorming stage tho so we can lay out ideas; loving the discussion so far.

**Comment by wtl420** [comment]
Classes and EX discussion should be moved to #6 now.

I also think we should think about integrating with other platforms so people can sign up with accounts elsewhere. Discord would be a prime target- I haven't seen any "sign in with Discord" things yet but I'm sure it's possible as Discord seems to have a pretty solid API.

Are there any other platforms we would want to think about? "Sign in with your MFGG account" maybe?
